rocAL/rocAL_pybind/examples/rocAL_api_coco_pipeline_segmentation.py

```diff
@@ -0,0 +1,555 @@
+
+    def fields(self):
+        return list(self.extra_fields.keys())
+
+    def _copy_extra_fields(self, bbox):
+        for k, v in bbox.extra_fields.items():
+            self.extra_fields[k] = v
+
+    def convert(self, mode):
+        if mode not in ("xyxy", "xywh"):
+            raise ValueError("mode should be 'xyxy' or 'xywh'")
+        if mode == self.mode:
+            return self
+        # we only have two modes, so don't need to check
+        # self.mode
+        xmin, ymin, xmax, ymax = self._split_into_xyxy()
+        if mode == "xyxy":
+            bbox = torch.cat((xmin, ymin, xmax, ymax), dim=-1)
+            bbox = BoxList(bbox, self.size, mode=mode)
+        else:
+            TO_REMOVE = 1
+            bbox = torch.cat(
+                (xmin, ymin, xmax - xmin + TO_REMOVE, ymax - ymin + TO_REMOVE), dim=-1
+            )
+            bbox = BoxList(bbox, self.size, mode=mode)
+        bbox._copy_extra_fields(self)
+        return bbox
+
+    def _split_into_xyxy(self):
+        if self.mode == "xyxy":
+            xmin, ymin, xmax, ymax = self.bbox.split(1, dim=-1)
+            return xmin, ymin, xmax, ymax
+        elif self.mode == "xywh":
+            TO_REMOVE = 1
+            xmin, ymin, w, h = self.bbox.split(1, dim=-1)
+            return (
+                xmin,
+                ymin,
+                xmin + (w - TO_REMOVE).clamp(min=0),
+                ymin + (h - TO_REMOVE).clamp(min=0),
+            )
+        else:
+            raise RuntimeError("Should not be here")
+
+
+class ROCALCOCOIterator(object):
+    """
+    COCO ROCAL iterator for pyTorch.
+    Parameters
+    ----------
+    pipelines : list of amd.rali.pipeline.Pipeline
+                List of pipelines to use
+    size : int
+           Epoch size.
+    """
+
+    def __init__(self, pipelines, tensor_layout=types.NCHW, reverse_channels=False, multiplier=None, offset=None, tensor_dtype=types.FLOAT16, device="cpu", display=False):
+
+        assert pipelines is not None, "Number of provided pipelines has to be at least 1"
+
+        self.loader = pipelines
+        self.tensor_format = tensor_layout
+        self.multiplier = multiplier if multiplier else [1.0, 1.0, 1.0]
+        self.offset = offset if offset else [0.0, 0.0, 0.0]
+        self.reverse_channels = reverse_channels
+        self.tensor_dtype = tensor_dtype
+        self.device = device
+        self.device_id = self.loader._device_id
+        self.bs = self.loader._batch_size
+        self.w = self.loader.getOutputWidth()
+        self.h = self.loader.getOutputHeight()
+        self.n = self.loader.getOutputImageCount()
+        self.rim = self.loader.getRemainingImages()
+        self.display = display
+        print("____________REMAINING IMAGES____________:", self.rim)
+        color_format = self.loader.getOutputColorFormat()
+        self.p = (1 if color_format is types.GRAY else 3)
+        if tensor_layout == types.NCHW:
+            if self.device == "cpu":
```

Comment:
please combine both cpu and gpu cases into single line 
device = None if device==cpu else torch_gpu_device